### PR TITLE
[DOCS] Clearing mappings

### PR DESCRIPTION
**Pull request by leonardoalt**
Security consideration about clearing arrays of mappings.

### CODE REVIEW

File: docs/security-considerations.rst [hunk 1]
```diff
@@ -254,6 +254,58 @@ if you want all overflows to cause a revert.
...
+.. _clearing-mappings:
+
+Clearing Mappings
+=================
+
+The Solidity type ``mapping`` (see :ref:`mapping-types`) is a storage-only key-value data structure that
+does not keep track of the keys that were assigned a non-zero value.
+Because of that, cleaning a mapping without extra information about the written
+keys is not possible.
+If a ``mapping`` is used as the base type of a dynamic storage array, deleting
```
Comment (chriseth) [inline docs/security-considerations.rst:266]: the mapping is also ignored in assignments of structs or arrays containing mappings.

File: docs/security-considerations.rst [hunk 2]
```diff
@@ -254,6 +254,58 @@ if you want all overflows to cause a revert.
...
+``writeMap(4, 128, 256)``.
+At this point, calling ``readMap(4, 128)`` returns 256.
+If we call ``eraseMaps``, the length of state variable ``array`` is zeroed, but
+since its ``mapping`` elements cannot be zeroed, their information stays alive
+in the contract's storage.
+After deleting ``array``, calling ``allocate(5)`` allows us to access
+``array[4]`` again, and calling ``readMap(4, 128)`` returns 256 even without
+another call to ``writeMap``.
+
+If your ``mapping`` information must be deleted, consider using an `iterable mapping <https://github.com/ethereum/dapp-bin/blob/master/library/iterable_mapping.sol>`_,
```
Comment (ChrisChinchilla) [inline docs/security-considerations.rst]: @leonardoalt This example is quite old, is it still best practice / current?

Comment (leonardoalt) [reply docs/security-considerations.rst]: Just by looking at it I'd say it works. I'll rephrase this sentence though.

**Comment by leonardoalt** [comment]
@ChrisChinchilla @chriseth updated

**Comment by chriseth** [comment]
Looks good! Please squash.

**Comment by leonardoalt** [comment]
Squashed

**Review by chriseth — APPROVED** [review]

**Pull request merged**